Std/Data/BitVec/Lemmas.lean

```diff
@@ -499,3 +511,42 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   let ⟨y, lt⟩ := y
   simp
   exact Nat.lt_of_le_of_ne
+
+-- YESUPSTREAMABLE
+@[simp]
+theorem BitVec.ofInt_ofNat : BitVec.ofInt w (no_index (OfNat.ofNat n : Int)) = BitVec.ofNat w n := by
+  simp [BitVec.ofInt]
+
+-- YESUPSTREAMABLE.
+-- for completeness, also do the signed versions with `toInt` and the other unsigned versions.
+theorem BitVec.ult_ofNat (w : Nat) (x y : Nat)  :
+    (BitVec.ult (BitVec.ofNat w x) (BitVec.ofNat w y)) = decide (x % 2^w < y % 2^w) := by
+  simp [BitVec.ult]
+  constructor
+  · simp [BitVec.ofNat, BitVec.toFin, Fin.ofNat']
+  · intros h
+    simp only [BitVec.ofNat, Fin.ofNat', Fin.mk_lt_mk, h]
+
+-- UPSTREAMABLE
+/- Not a simp lemma by default because we may want toFin or toInt in the future. -/
+theorem BitVec.ult_toNat (x y : BitVec n) :
+    (BitVec.ult (n := n) x y) = decide (x.toNat < y.toNat) := by
+    have h : x = BitVec.ofNat n x.toNat := by simp
+    have h' : y = BitVec.ofNat n y.toNat := by simp
+    rw [h, h']
+    apply BitVec.ult_ofNat
```

Comment:
I'm not sure we should upstream this if we have `ult_ofNat`. Feel free to put it in the Backlog, though